airbyte-webapp/src/components/connectorBuilder/types.ts

```diff
@@ -231,12 +276,47 @@ export const builderFormValidationSchema = yup.object().shape({
       name: yup.string().required("form.empty.error"),
       urlPath: yup.string().required("form.empty.error"),
       fieldPointer: yup.array().of(yup.string()),
+      cursorField: yup.array().of(yup.string()),
+      primaryKey: yup.array().of(yup.string()),
       httpMethod: yup.mixed().oneOf(["GET", "POST"]),
       requestOptions: yup.object().shape({
         requestParameters: yup.array().of(yup.array().of(yup.string())),
         requestHeaders: yup.array().of(yup.array().of(yup.string())),
         requestBody: yup.array().of(yup.array().of(yup.string())),
       }),
+      paginator: yup
+        .object()
+        .shape({
+          pageSizeOption: requestOptionSchema,
+          pageTokenOption: requestOptionSchema,
+          strategy: yup
+            .object({
+              page_size: yup.mixed().when("type", {
+                is: (val: string) => ["OffsetIncrement", "PageIncrement"].includes(val),
+                then: yup.number().required("form.empty.error"),
+                otherwise: yup.number(),
+              }),
+              cursor_value: yup.mixed().when("type", {
+                is: "CursorPagination",
+                then: yup.string().required("form.empty.error"),
+                otherwise: (schema) => schema.strip(),
+              }),
+              stop_condition: yup.mixed().when("type", {
+                is: "CursorPagination",
+                then: yup.string(),
+                otherwise: (schema) => schema.strip(),
+              }),
+              start_from_page: yup.mixed().when("type", {
+                is: "PageIncrement",
+                then: yup.string(),
+                otherwise: (schema) => schema.strip(),
+              }),
+            })
+            .notRequired()
+            .default(undefined),
```

Comment:
Without this, the UI was failing when toggling on the paginator component I think because the values are not all initialized

Reply:
yup is getting more annoying by the day as well.... I think it's fine for now, but I would really like to just leverage the json schema of the manifest itself.

Reply:
100% agree. It was a pretty unfun and tedious process to go through and recreate this manually, and it is not very readable